### PR TITLE
Issue 78 emojipicker ref

### PR DESCRIPTION
**Pull request by faebebin**
I also work on this issue because I came accross the same topic working on the table button. Problem 'update component (ie. emojibutton) upon change of ref state'. Solution found here: https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node . For consistency reasons I also implemented it here.

### CODE REVIEW

File: packages/editor/src/client/atoms/emojiButton.tsx [hunk 1]
```diff
@@ -67,17 +33,25 @@ export function EmojiButton({icon, iconActive}: EmojiButtonProps) {
...
+            padding: 2
+          }}
           onMouseDown={e => {
             e.preventDefault()
-            isEmojiPickerOpen ? close() : open()
-          }}
-        />
+            !isPopoverOpen ? triggerRef.current!.open() : triggerRef.current!.close()
+          }}>
+          <Icon icon={icon} element={icon} />
```
Comment (roosnic1) [inline packages/editor/src/client/atoms/emojiButton.tsx:53]: do you need the property `element`?

**Review by roosnic1 — COMMENTED** [review]
looks good. just one question.

File: packages/editor/src/client/atoms/emojiButton.tsx (hunk 1 above)
Comment (faebebin) [reply packages/editor/src/client/atoms/emojiButton.tsx]: No :+1:

File: packages/editor/src/client/atoms/emojiButton.tsx [hunk 2]
```diff
@@ -11,54 +11,20 @@ import {Picker, BaseEmoji} from 'emoji-mart'
...
-    }
-  }
-  const close = () => {
-    if (triggerRef.current) {
-      triggerRef.current!.close()
-      setIsEmojiPickerOpen(false)
-    }
-  }
+  const popOverRef = useCallback((node: any) => {
+    setIsPopoverOpen(!!node)
```
Comment (0815pascal) [inline packages/editor/src/client/atoms/emojiButton.tsx:23]: Do the `!!` have a special role? Otherwise `!!node === node` wouldn't it?

**Review by 0815pascal — APPROVED** [review]
Looks good, just one question

**Review by roosnic1 — APPROVED** [review]

File: packages/editor/src/client/atoms/emojiButton.tsx (hunk 2 above)
Comment (faebebin) [reply packages/editor/src/client/atoms/emojiButton.tsx:23]: I actually found this on this repo. It turns any defined object into boolean, which is needed in this case.

Comment (0815pascal) [reply packages/editor/src/client/atoms/emojiButton.tsx:23]: Ah ok, I didn't know that - just learned something new :)

**Pull request merged**